conan.cmake

```diff
@@ -64,28 +70,11 @@ function(conan_cmake_settings result)
       set(_SETTINGS ${_SETTINGS} -s compiler=clang -s compiler.version=${MAJOR}.${MINOR} -s compiler.libcxx=libstdc++)
   elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL MSVC)
       set(_VISUAL "Visual Studio")
-      if (MSVC_VERSION EQUAL 1200)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=6)
-      elseif (MSVC_VERSION EQUAL 1300)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=7)
-      elseif (MSVC_VERSION EQUAL 1310)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=7)
-      elseif (MSVC_VERSION EQUAL 1400)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=8)
-      elseif (MSVC_VERSION EQUAL 1500)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=9)
-      elseif (MSVC_VERSION EQUAL 1600)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=10)
-      elseif (MSVC_VERSION EQUAL 1700)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=11)
-      elseif (MSVC_VERSION EQUAL 1800)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=12)
-      elseif (MSVC_VERSION EQUAL 1900)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=14)
-      elseif (MSVC_VERSION EQUAL 1910)
-          set(_SETTINGS ${_SETTINGS} -s compiler=${_VISUAL} -s compiler.version=15)
-      else ()
+      _get_msvc_version(_VISUAL_VERSION)
+      if("{_VISUAL_VERSION}" STREQUAL "")
```

Comment:
What variable do you want, `msvc_version`? I choosed `_VISUAL_VERSION` only because your other variable was called `_VISUAL`.